### DISCUSSION TITLE
Splitting transactions into separate edi files

### DISCUSSION THREAD

**Discussion by hancockks** [xlate/staedi#387] [Q&A] [answered]
I currently have a prototype using the EDIStreamReader to build up an EDI pojo at the ICE/GROUP/Transaction level.

Since I don't need to break down anything beyond a list of segments within each transaction, is there a way to configure the stream reader to just read a composite as a single segment text section?

**Comment by MikeEdgar** [comment, marked as answer]
> Since I don't need to break down anything beyond a list of segments within each transaction, is there a way to configure the stream reader to just read a composite as a single segment text section?

Hi @hancockks , do you need to just list the segment tags or are you also using the element data?

To answer your question, there isn't any configuration option for that, but you could easily use a `StringBuilder` for it. When your application encounters a `START_COMPOSITE` event, set the length of the builder to 0. Then, for each `ELEMENT_DATA` event, append the text from the event. Finally, use the value from the builder when `END_COMPOSITE` occurs.

**Reply by hancockks** [reply]
I'm going to end up outputting multiple files with the same ICE, same group (changing the GE to have one group), so while I'm not looking at the individual segments, I'll need to capture everything within the transaction as-is.  The StringBuilder approach is fine.

Can you think of any other gotchas other than the composites?  They'll always be text ELEMENT_DATA sections? (no possibility of binary?)
